main.js

```diff
@@ -153,31 +153,34 @@ getCookie = function(request, cookie) {
 
 app.get('/confirm', function(req, res){
   if (isUaaSession(req)){
-    var confirmParams = 'client_id=' + req.session.client_id
-                        + '&response_type=' + req.session.response_type
-                        + '&scope=' + req.session.scope;
-                        + '&redirect_uri=' + req.session.redirect_uri;
+    var confirmData = {
+        'client_id' : req.session.client_id,
+        'response_type': req.session.response_type,
+        'scope' : req.session.scope,
+        'redirect_uri' : req.session.redirect_uri
+    }
     var confirmOptions = {
                       headers: {
+                        'Accept' : 'application/json',
                         'Cookie': 'JSESSIONID=' + getCookie(req, 'uaa_cookie')
                          }
                   }
-    needle.get(uaaAddress + '/oauth/authorize?' + confirmParams, confirmOptions,
-        function(err, confirmResp) {
-            if (confirmResp.statusCode == 200){
-                res.cookie('JSESSIONID', getCookie(req, 'uaa_cookie'))
-                res.render('confirm', {client_id : req.session.client_id})
-            } else if (confirmResp.statusCode == 302){
-                if (endsWith(confirmResp.headers.location, '/login')){ // when redirects to UAA API login page
-                  res.render('login',{ errorMessage: "" });
-                } else {
-                  res.cookie('JSESSIONID', getCookie(req, 'uaa_cookie'))
-                  res.redirect(confirmResp.headers.location)
-                }
-            } else {
-                res.end('Login/confirm: Error from token server, code: ' + confirmResp.statusCode)
-            }
-        });
+     needle.post(uaaAddress + '/oauth/authorize', confirmData, confirmOptions, function (err, confirmResp){
+                    if (confirmResp.statusCode == 200){
+                        req.session.userScopes = confirmResp.body.auth_request.scope
+                        res.cookie('JSESSIONID', getCookie(req, 'uaa_cookie'))
+                        res.render('confirm', {client_id : req.session.client_id})
+                    } else if (confirmResp.statusCode == 302){
+                        if (endsWith(confirmResp.headers.location, '/login')){ // when redirects to UAA API login page
+                          res.render('login',{ errorMessage: "" });
+                        } else {
+                          res.cookie('JSESSIONID', getCookie(req, 'uaa_cookie'))
+                          res.redirect(confirmResp.headers.location)
+                        }
+                    } else {
+                        res.end('Login/confirm: Error from token server, code: ' + confirmResp.statusCode)
```

Comment:
we should log the status code/error message here because we won't be able to track what the error was